### PR TITLE
doc: update README to align with the new project brand

### LINKED ISSUE #150: Update the README for the repository
### Is your feature request related to a problem?

I propose to update the README.md file for the repository with the following information:
- link to the overview of the organization
- purpose of the repository
- is it in active development or not, is it archived or not, any reasons for that
- links to relevant documentation (security audits, web site, design documents, etc.)

### What solution do you propose?

N/A

### What alternatives have you considered?

N/A

### Any additional context?

_No response_

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Resolve #150

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,28 @@
-# notation-core-go
+# Notation-core-go library
```
Comment (yizha1) [inline README.md:1]: Seems a different pattern from the `notation-go` readme. Maybe we can try `Notation Core Golang Library`

File: README.md [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+## Table of Contents
+- [Core Documents](#core-documents)
+- [Code of Conduct](#code-of-conduct)
+- [License](#license)
+ 
+## Core Documents
+
+All library documentation is available at [Go Reference](https://pkg.go.dev/github.com/notaryproject/notation-core-go).
+
+## Code of Conduct
```
Comment (yizha1) [inline README.md:22]: Suggest referencing to `https://github.com/notaryproject/.github/blob/main/CODE_OF_CONDUCT.md`, which was updated months ago per TOC review.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
 [![Build Status](https://github.com/notaryproject/notation-core-go/actions/workflows/build.yml/badge.svg?event=push&branch=main)](https://github.com/notaryproject/notation-core-go/actions/workflows/build.yml?query=workflow%3Abuild+event%3Apush+branch%3Amain)
 [![codecov](https://codecov.io/gh/notaryproject/notation-core-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notaryproject/notation-core-go)
 [![Go Reference](https://pkg.go.dev/badge/github.com/notaryproject/notation-core-go.svg)](https://pkg.go.dev/github.com/notaryproject/notation-core-go@main)
 
-Contains support for Notary Project signature envelope, and format specific implementation
+Notation-core-go library provides specific implementation for Notary Project signature envelope, signature format, and OCSP revocation check.
+
+Notation-core-go library reaches stable in July 2023 and is in active development status.
+
+> **Note** The Notation documentation is available [here](https://notaryproject.dev/docs/). You can also find the Notary Project [README](https://github.com/notaryproject/.github/blob/main/README.md) to learn about the overall Notary Project.
```
Comment (yizha1) [inline README.md:11]: Suggest moving overall intro to line 10

**Review by yizha1 — COMMENTED** [review]
Thanks @FeynmanZhou a few comments

File: README.md [hunk 4]
```diff
@@ -1,7 +1,28 @@
-# notation-core-go
+# Notation-core-go library
 
 [![Build Status](https://github.com/notaryproject/notation-core-go/actions/workflows/build.yml/badge.svg?event=push&branch=main)](https://github.com/notaryproject/notation-core-go/actions/workflows/build.yml?query=workflow%3Abuild+event%3Apush+branch%3Amain)
 [![codecov](https://codecov.io/gh/notaryproject/notation-core-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notaryproject/notation-core-go)
 [![Go Reference](https://pkg.go.dev/badge/github.com/notaryproject/notation-core-go.svg)](https://pkg.go.dev/github.com/notaryproject/notation-core-go@main)
 
-Contains support for Notary Project signature envelope, and format specific implementation
+Notation-core-go library provides specific implementation for Notary Project signature envelope, signature format, and OCSP revocation check.
```
Comment (Two-Hearts) [inline README.md:7]: nit:
```suggestion
notation-core-go library provides specific implementation for Notary Project signature envelope, signature format, and OCSP revocation check.
```

File: README.md [hunk 5]
```diff
@@ -1,7 +1,28 @@
...
+
+Notation-core-go library reaches stable in July 2023 and is in active development status.
```
Comment (Two-Hearts) [inline README.md:9]:
```suggestion
notation-core-go library reaches stable in July 2023 and is in active development status.
```

File: README.md (hunk 1 above)
Comment (Two-Hearts) [inline README.md:1]: Why are we making this change? I think we can keep the title to be same as the repo's name, i.e. `notation-core-go`.

File: README.md (hunk 4 above)
Comment (priteshbandi) [inline README.md:7]:
```suggestion
notation-core-go implements signature generation, parsing and revocation related functionalities based on [Notary Project specifications](link).
```

File: README.md [hunk 6]
```diff
@@ -1,7 +1,28 @@
...
+
+> **Note** The Notation documentation is available [here](https://notaryproject.dev/docs/). You can also find the Notary Project [README](https://github.com/notaryproject/.github/blob/main/README.md) to learn about the overall Notary Project.
+
+## Table of Contents
+- [Core Documents](#core-documents)
+- [Code of Conduct](#code-of-conduct)
+- [License](#license)
+ 
+## Core Documents
```
Comment (priteshbandi) [inline README.md:18]:
```suggestion
## Documentation
```

**Review by Two-Hearts — APPROVED** [review]
LGTM

File: README.md [hunk 7]
```diff
@@ -4,4 +4,27 @@
...
+notation-core-go implements signature generation, parsing and revocation related functionalities based on the [Notary Project specifications](https://github.com/notaryproject/specifications).
+
+notation-core-go reached a stable release as of July 2023 and continues to be actively developed.
+
+You can also find the Notary Project [README](https://github.com/notaryproject/.github/blob/main/README.md) to learn about the overall Notary Project.
+
+> **Note** The Notary Project documentation is available [here](https://notaryproject.dev/docs/).
+
+## Table of Contents
+- [Core Documents](#core-documents)
```
Comment (priteshbandi) [inline README.md:16]:
```suggestion
- [Documentation](#documentation)
```

File: README.md [hunk 8]
```diff
@@ -4,4 +4,27 @@
 [![codecov](https://codecov.io/gh/notaryproject/notation-core-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notaryproject/notation-core-go)
 [![Go Reference](https://pkg.go.dev/badge/github.com/notaryproject/notation-core-go.svg)](https://pkg.go.dev/github.com/notaryproject/notation-core-go@main)
 
-Contains support for Notary Project signature envelope, and format specific implementation
+notation-core-go implements signature generation, parsing and revocation related functionalities based on the [Notary Project specifications](https://github.com/notaryproject/specifications).
```
Comment (Two-Hearts) [inline README.md:7]: nit:
notation-core-go also provides `certificate and certificate chain validation`.

Comment (shizhMSFT) [inline README.md:7]: Should emphasize on that `notation-core-go` provides core crypto functionality for `notation-go`.

File: README.md [hunk 9]
```diff
@@ -4,4 +4,27 @@
...
+
+notation-core-go reached a stable release as of July 2023 and continues to be actively developed.
```
Comment (shizhMSFT) [inline README.md:9]: developed **and maintained**?

File: README.md [hunk 10]
```diff
@@ -4,4 +4,27 @@
...
+
+You can also find the Notary Project [README](https://github.com/notaryproject/.github/blob/main/README.md) to learn about the overall Notary Project.
+
+> **Note** The Notary Project documentation is available [here](https://notaryproject.dev/docs/).
```
Comment (shizhMSFT) [inline README.md:13]:
```suggestion
> [!NOTE]
> The Notary Project documentation is available [here](https://notaryproject.dev/docs/).
```
> [!NOTE]
> The Notary Project documentation is available [here](https://notaryproject.dev/docs/).

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by yizha1 — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Comment by FeynmanZhou** [comment]
@priteshbandi @patrickzheng200 This is PR is ready to be merged. Would you mind doing it? Thanks

**Pull request merged**